packages/blocks/src/api/registration.js

```diff
@@ -98,10 +101,23 @@ export function registerBlockType( name, settings ) {
 		return;
 	}
 
+	const preFilterSettings = { ...settings };
 	settings = applyFilters( 'blocks.registerBlockType', settings, name );
 
 	if ( settings.deprecated ) {
-		settings.deprecated = settings.deprecated.map( ( deprecation ) => applyFilters( 'blocks.registerBlockType', deprecation, name ) );
+		settings.deprecated = settings.deprecated.map( ( deprecation ) =>
+			pick(
+				applyFilters(
+					'blocks.registerBlockType',
+					{
```

Comment:
It could maybe do for an inline comment explaining the purpose of the merge, `omit`, `pick`.

Reply:
On it!